### PR TITLE
Add more XEP infos to Settingsdialog

### PR DESCRIPTION
**Pull request by mightymop**
improved css for settings dialog
added tabs ... 

I took the XEPs from jsxc.org website, i did not proof the code for it...
So if all upcoming changes would be merged the xeps should be updated then...

### CODE REVIEW

File: scss/partials/_dialog.scss [hunk 1]
```diff
@@ -160,6 +160,73 @@
   }
 }
 
+.jsxc-tabbed-area figure {
```
Comment (sualko) [inline scss/partials/_dialog.scss:163]: Please the nesting capabilities of sass.

File: scss/partials/_dialog.scss [hunk 2]
```diff
@@ -160,6 +160,73 @@
...
+.jsxc-tabbed-area > input,
+.jsxc-tabbed-area figure > section { display: none; }
+
+.jsxc-tabbed-area figure>section {
+  background: var(--jsxc-color-main-background);
+  border: 1px solid var(--jsxc-color-border);
+  color: var(--jsxc-color-text-light);
+  letter-spacing: 0.3px;
+  line-height: 1.5em;
+  margin-top: 0 !important;
```
Comment (sualko) [inline scss/partials/_dialog.scss:183]: The use of important often indicates some issues with the html or css structure.

File: src/ui/dialogs/settings.ts [hunk 3]
```diff
@@ -109,19 +109,107 @@ class AccountPage extends Page {
...
    protected generateContentElement(): JQuery {
-      let contentElement = $('<div>');
 
-      contentElement.append(new ConnectionSection(this.navigation, this.account).getDOM());
-      contentElement.append(new PluginSection(this.navigation, this.account).getDOM());
+      let contentElement = $('<div class="jsxc-tabbed-area">');
+
+      let tabconnection = $('<input checked="checked" id="jsxc-tabbed-area-tab1" type="radio" name="tabs" />');
+      let tabmainapp = $('<input  id="jsxc-tabbed-area-tab2" type="radio" name="tabs" />');
+      let tabplugins = $('<input  id="jsxc-tabbed-area-tab3" type="radio" name="tabs" />');
```
Comment (sualko) [inline src/ui/dialogs/settings.ts:117]: Should be abstracted so we don't have to copy code if we add another tabular.

File: src/ui/dialogs/settings.ts [hunk 4]
```diff
@@ -261,7 +349,15 @@ class PluginSection extends Section {
...
          checkboxElement.attr('id', id);
          checkboxElement.attr('name', name);
+
          checkboxElement.prop('checked', disabledPlugins.indexOf(id) < 0);
+
+         if (id==='omemo'&&typeof (<any> window).libsignal === 'undefined')
+         {
+          checkboxElement.prop('checked',false);
+          checkboxElement.prop('disabled',true);
+         }
```
Comment (sualko) [inline src/ui/dialogs/settings.ts:359]: :+1:

File: src/ui/dialogs/settings.ts [hunk 5]
```diff
@@ -281,18 +377,21 @@ class PluginSection extends Section {
 
          if (Array.isArray(metaData.xeps)) {
             metaData.xeps.forEach(xep => {
-               let xepElement = $('<a>');
+               let xepElement = $('<a target="_blank">');
```
Comment (sualko) [inline src/ui/dialogs/settings.ts:380]: We need `rel="noreferrer noopener"`.

**Review by sualko — CHANGES_REQUESTED** [review]
In addition to the new design, the pr also includes some important bug fixes. Would be nice if there would be a commit for each of them, so we don't forget to mention them in the upcoming changelog.

2021-01-05 13 44 11 localhost 8f2c770453b4

I don't like the scrollbar inside the dialog.

2021-01-05 13 44 45 localhost f45917537172

On small screens the view breaks.

I see the need to enhance the view with a growing number of plugins and the non-plugin xeps added, but maybe it's easier to make some sections collapsible. This way the code has to be changed only a bit and we don't have to expect any mobil issues.

**Comment by mightymop** [comment]
ok i will create a new branch in your repo with extra commits for the bugfixes and collapsibles..

**Comment by mightymop** [comment]
This could then be closed now for this PR: https://github.com/jsxc/jsxc/pull/992

**Pull request closed without merging**